hummingbot/hummingbot
# Fix/kucoin get fees multiple pairs

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
Fix in KucoinAuth when preparing the string to generate the signature. There was a problem processing the request parameters when any of the parameters contained a comma separated list.
The PR also includes a fix to a couple of CoinFlex unit tests that were failing.


**Tests performed by the developer**:
Added a new unit test with the same scenario from the issue.
Tested with a script that updated the fees on a Kucoin connector instance configured to use more than one trading pair.

**Test summary**:
- Installed the feature branch without any issue
- Run a liquidity mining strategy with multiple markets for more than 6 hrs
- Reviewed the logs and confirmed that the error with the fee is no longer seen
- Check the fees paid after a filled order took place and confirmed they are correct.
- Build a docker image and create a docker instances
- Repeat the same step done on source build and confirm not seeing the fee error.

**Tips for QA testing**:
To test the fix it is required to use a strategy that works with more than one trading pair.


Fixes https://github.com/hummingbot/hummingbot/issues/5273

This PR comes from https://github.com/CoinAlpha/hummingbot/pull/253

PRP: https://snapshot.org/#/hbot-prp.eth/proposal/0x4d9ee5f562c4b377bc30b3799cc869191dec4a2abe6db66cf1d954a84e5f13e5
